### PR TITLE
Handle multiple dashes in phone number detection in english

### PR DESCRIPTION
**Pull request by debadityamandal**
This commit will replace em-dash and en-dash with hyphen which will help us to find the locale easily in phone number. Since all dashes look like same, we can't separate them by any expression because except hyphen another dashes are not available in keyboard but their unicode values are different. To do this I replaced re module by regex module.

### CODE REVIEW

File: ner_v2/detectors/pattern/phone_number/phone_number_detection.py [hunk 1]
```diff
@@ -50,7 +50,8 @@ def get_country_code_from_locale(self):
         """
         This method sets self.country_code from given locale
         """
-        regex_pattern = re.compile('[-_](.*$)', re.U)
+        regex_pattern = regex.compile('[-_](.*$)', regex.U)
+        self.locale=regex.sub("\\p{Pd}","-",self.locale)  # This will replace all types of dashes(em or en) by hyphen.
```
Comment (haptik-deployment) [inline ner_v2/detectors/pattern/phone_number/phone_number_detection.py:54]: E225 missing whitespace around operator
E231 missing whitespace after ','

File: ner_v2/detectors/pattern/phone_number/phone_number_detection.py [hunk 2]
```diff
@@ -123,12 +124,12 @@ def check_for_country_code(self, phone_num):
         phone_dict = {}
 
         if len(phone_num) > 10:
-            check_country_regex = re.compile(r'^({country_code})\d{length}$'.
-                                             format(country_code='911|1|011 91|91', length='{10}'), re.U)
+            check_country_regex = regex.compile(r'^({country_code})\d{length}$'.
+                                             format(country_code='911|1|011 91|91', length='{10}'), regex.U)
```
Comment (haptik-deployment) [inline ner_v2/detectors/pattern/phone_number/phone_number_detection.py:128]: E128 continuation line under-indented for visual indent

**Comment by haptik-unit-test** [comment]
Can one of the admins verify this patch?

**Comment by haptik-deployment** [comment]
:+1: No lint errors found.

**Comment by chiragjn** [comment]
Hello, thank you for your contribution. Sorry for the late reply :sweat_smile: 
Can you please give us an example case of `locale` where such kind of substitution is needed?
Normally because `locale` is supposed to be passed from the outside, we would normally expect it to be correct

**Comment by debadityamandal** [comment]
I was using this module to extract phone number from few documents. That time I got locale "EN–US" from those documents and I got the following response-
([{'country_calling_code': '91', 'value': 'XXXXXXXXXX'}], ['XXXXXXXXXX'])
This 'country_calling_code' is representing India but I gave US. Then I started to debug the code and I saw it was not detecting country code because of the wrong dash as the code is detecting only hyphen. We can't type Em-Dash/En-Dash through keyboard normally. So, I converted the Em-Dash which is mentioned in locale to hyphen and now I am getting following response-
([{'country_calling_code': '1', 'value': 'XXXXXXXXXX'}], ['XXXXXXXXXX']).

**Comment by chiragjn** [comment]
Although this seems like something that should be handled outside of this module, I am still willing to accept the change. Can you please revert all `re` changes? We would like to use both `re` and `regex` (only for the dash replacing use cases) for now to keep the diff as small as possible

**Comment by haptik-deployment** [comment]
:+1: No lint errors found.

**Comment by debadityamandal** [comment]
I revert all **re** changes. Please let me know if you have any doubt.

**Comment by haptik-deployment** [comment]
:+1: No lint errors found.

**Review by chiragjn — APPROVED** [review]

**Comment by debadityamandal** [comment]
Thank you for accepting my code. I am closing pull request

**Pull request merged**